src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -59,6 +94,9 @@ client_ptr kafka_client_cache::fetch_or_insert(
                 vlog(plog.debug, "Cache size reached, evicting {}", item.key);
                 inner_list.pop_back();
                 _evicted_items.push_back(std::move(item));
+                // Immediately run the eviction process
+                _eviction_timer.rearm(
+                  ss::timer<ss::lowres_clock>::time_point(0s));
```

Comment:
Does this work? I'm unsure about how the units are converted, but I think `0s` is added to the epoch. Perhaps `ss::timer<ss::lowres_clock>::now()` is what you wanted.

I think the comment is slightly misleading, since immediately implies synchronously. Perhaps "trigger" or "in the background" might be clearer.